### DISCUSSION TITLE
Sweeper or Voice Track over song

### DISCUSSION THREAD

**Discussion by DJMic** [savonet/liquidsoap#3939] [Q&A]
Is there a way to play some sweepers or voice track at the begining of a song, like professional radio station ?
Is it possible to define "Intro Point" ?

**Comment by Moonbase59** [comment]
I don’t think that’s easily possible with the current versions, in an automated way—correct me if I’m wrong.

Nevertheless, `cue_file` (part of my autocue implementation) has already reserved some metadata names for future ramp (intro) and hook points:
```
    # reserved for future expansion
    "liq_ramp1": float,
    "liq_ramp2": float,
    "liq_ramp3": float,
    "liq_hook1_in": float,
    "liq_hook1_out": float,
    "liq_hook2_in": float,
    "liq_hook2_out": float,
    "liq_hook3_in": float,
    "liq_hook3_out": float,
```

I hope AzuraCast’s visual cue editor will one day support setting at least `liq_ramp1` and maybe `liq_hook1_in` and `liq_hook1_out` which could simplify automation a lot.

Using `liq_ramp1`, we could at least easily determine if a sweeper/liner "fits" over the intro, before vocals set in, and use existing Liquidsoap features (like `add`, `smooth_add`) to play it "on top":

![Voice_tracking_talkover_example](https://github.com/savonet/liquidsoap/assets/3706922/9ac956f0-7d18-44f7-aeab-994dd3026f88)

The problem with these is that each and every radio automation and DJing software uses their own tags for these, because they’re not standardized. Thus my early proposal. If Liquidsoap uses it, maybe others pick it up…
